tests/transports/test_all_plugins.py

```diff
@@ -1358,10 +1362,135 @@ def test_exec_with_stdin_filelike(self, custom_transport):
+        """
+        test_string = b'some_test bytes with non-unicode -> \xFA'
+        stdin = io.BytesIO(test_string)
+        with custom_transport as transport:
+            retcode, stdout, stderr = transport.exec_command_wait_bytes('cat', stdin=stdin)
+            self.assertEqual(retcode, 0)
+            self.assertEqual(stdout, test_string)
+            self.assertEqual(stderr, b'')
+
+    @run_for_all_plugins
+    def test_exec_with_stdin_filelike_bytes_decoding(self, custom_transport):
+        """Test command execution with a stdin from filelike of type bytes, trying to decod.
+
+        I test that the exec_command_wait function fails if I pass some non-unicode
+        bytes to check that there is no internal implicit encoding/decoding in the code.
+        """
+        test_string = b'some_test bytes with non-unicode -> \xFA'
+        stdin = io.BytesIO(test_string)
+        with custom_transport as transport:
+            with self.assertRaises(UnicodeDecodeError):
+                transport.exec_command_wait('cat', stdin=stdin, encoding='utf-8')
+
     @run_for_all_plugins
     def test_exec_with_wrong_stdin(self, custom_transport):
         """Test command execution with incorrect stdin string."""
         # I pass a number
         with custom_transport as transport:
             with self.assertRaises(ValueError):
                 transport.exec_command_wait('cat', stdin=1)
+
+    @run_for_all_plugins
+    def test_transfer_big_stdout(self, custom_transport):  # pylint: disable=too-many-locals
+        """Test the transfer of a large amount of data on stdout."""
+        import os
+        import random
+        import string
+        import tempfile
+
+        # Create a "big" file of > 2MB (10MB here; in general, larger than the buffer size)
+        min_file_size_bytes = 5 * 1024 * 1024
+        # The file content will be a sequence of these lines, until the size
+        # is > MIN_FILE_SIZE_BYTES
+        file_line = 'This is a Unicødê štring\n'
+        fname = 'test.dat'
+        script_fname = 'script.py'
+
+        # I create a large content of the file (as a string)
+        file_line_binary = file_line.encode('utf8')
+        line_repetitions = (min_file_size_bytes // len(file_line_binary) + 1)
+        fcontent = (file_line_binary * line_repetitions).decode('utf8')
+
+        with custom_transport as trans:
+            # We cannot use tempfile.mkdtemp because we're on a remote folder
+            location = trans.normalize(os.path.join('/', 'tmp'))
+            directory = 'temp_dir_test_transfer_big_stdout'
+            trans.chdir(location)
+
+            self.assertEqual(location, trans.getcwd())
+            while trans.isdir(directory):
+                # I append a random letter/number until it is unique
+                directory += random.choice(string.ascii_uppercase + string.digits)
+            trans.mkdir(directory)
+            trans.chdir(directory)
```

Comment:
This feels a bit zigzaggy, it took me a while to understand what was happening. Perhaps this is more straightforward to understand:

```python
            location = trans.normalize(os.path.join('/', 'tmp'))
            trans.chdir(location)
            self.assertEqual(location, trans.getcwd())

            directory = 'temp_dir_test_transfer_big_stdout
            while trans.isdir(directory):
                # I append a random letter/number until it is unique
                directory += random.choice(string.ascii_uppercase + string.digits)
            trans.mkdir(directory)
            trans.chdir(directory)
```